### PR TITLE
Topic/leela/falun ptx 20629

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Adding validations FA LUNS

**Which issue(s) this PR fixes** (optional)
Closes # PTX-20629

**Special notes for your reviewer**:
Result : https://aetos.pwx.purestorage.com/resultSet/testSetID/436515

### CODE REVIEW

File: pkg/pureutils/pure_types.go [hunk 1]
```diff
@@ -94,3 +108,260 @@ func GetPureSecret(output interface{}, name, namespace string) error {
 
 	return nil
 }
+
+func getPureFAVolumesJsonData(mgmtEndpoint, apiToken string) (string, error) {
```
Comment (kphalgun-px) [inline pkg/pureutils/pure_types.go:112]: Does it work for FB as well?

File: tests/basic/pure_test.go [hunk 2]
```diff
@@ -1862,3 +1943,231 @@ var _ = Describe("{CreateAndDeleteMultipleVolumesInParallel}", func() {
...
+					if err != nil {
+						errCh <- err
+						break
+					}
+					err = Inst().V.WaitDriverUpOnNode(stNode, 5*time.Minute)
+					if err != nil {
+						errCh <- err
+						break
+					}
+					break //remove it while raising PR
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:2081]: Comment to not forget

File: tests/basic/pure_test.go [hunk 3]
```diff
@@ -1862,3 +1943,231 @@ var _ = Describe("{CreateAndDeleteMultipleVolumesInParallel}", func() {
...
+			}(errCh, &wg)
+
+			wg.Wait()
+			close(errCh)
+
+			if len(errCh) > 0 {
+				for err := range errCh {
+					log.Errorf("%v", err)
+				}
+				log.FailOnError(fmt.Errorf("error(s) occured while creating PVC and restartign PX on nodes"), "no errors should occuer")
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:2092]: *restarting PX on nodes
no errors should *occur

File: tests/basic/pure_test.go [hunk 4]
```diff
@@ -1862,3 +1943,231 @@ var _ = Describe("{CreateAndDeleteMultipleVolumesInParallel}", func() {
...
+		}
+
+		for _, pvc := range pvcList.Items {
+			err := Inst().S.WaitForSinglePVCToBound(pvc.Name, nsName)
+			log.FailOnError(err, fmt.Sprintf("error validating PVC [%s] status in namespace [%s]", pvc.Name, nsName))
+		}
+
+		var faVols []string
+		for _, fa := range flashArrays {
+			v, err := pureutils.GetPureFAVolumes(fa.MgmtEndPoint, fa.APIToken)
```
Comment (kphalgun-px) [inline tests/basic/pure_test.go:2120]: Shouldn't we filter volumes that are created by this test case?

File: pkg/pureutils/pure_types.go (hunk 1 above)
Comment (lsrinivas-pure) [reply pkg/pureutils/pure_types.go:112]: should work but not tested

File: tests/basic/pure_test.go (hunk 2 above)
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:2081]: fixed

File: tests/basic/pure_test.go (hunk 3 above)
Comment (lsrinivas-pure) [reply tests/basic/pure_test.go:2092]: fixed

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Pull request merged**